schema.json

```diff
@@ -0,0 +1,214 @@
+{
+  "id": "http://api.codeclimate.com/test-coverage-schema#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "title": "Test Coverage",
+  "description": "Schema that describes the format of the test coverage payload sent to Code Climate",
+
+  "definitions": {
+    "covered_percent": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 100,
+    },
+    "covered_strength": {
+      "type": "integer",
+      "minimum": 0,
+    },
+
+    "line_counts": {
+      "type": "object",
+      "properties": {
+        "covered": {
+          "description": "Number of covered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "missed": {
+          "description": "Number of uncovered lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+        "total": {
+          "description": "Total lines of code",
+          "type": "integer",
+          "minimum": 0,
+        },
+      },
+      "required": ["covered", "missed", "total"],
+    },
+
+    "git_sha": {
+      "type": "string",
+      "pattern": "^\d{1,}\.\d{1,}\.\d{1,}",
+    },
+
+    "source_file": {
+      "description": "Meta-data and coverage information for source file under test.",
+      "type": "object",
+      "properties": {
+        "blob_id": {
+          "description": "Blob SHA as reported by git",
+          "$ref": "#/definitions/git_sha"
+        },
+        "coverage": {
+          "description": "Actual line by line coverage data, represented as a string for safety. A null represents not coverable, and a 0-n represents the number of times the line is covered.",
+          "type": "string",
+        },
+        "name": {
+          "description": "Path to source file under test",
+          "type": "string",
+        },
+        "covered_percent": {
+          "$ref": "#/definitions/covered_percent"
+        },
+        "covered_strength": {
+          "$ref": "#/definitions/covered_strength"
+        },
+        "line_counts": {
+          "$ref": "#/definitions/line_counts"
+        }
+      },
+      "required": ["name", "blob_id", "coverage"],
+    },
+
+    "environment": {
+      "description": "Information and context related to the environment where tests are run",
+      "type": "object",
+      "properties": {
+        "gem_version": {
+          "description": "Simplecov gem version",
```

Comment:
This was actually the version of the `gem` program. Someone thought it would be useful to know `gem --version` since our first test report was implemented as a gem.